code/lib/core-server/src/utils/__tests__/remove-mdx-stories.test.ts

```diff
@@ -0,0 +1,274 @@
+
+  await expect(
+    removeMDXEntries(
+      Object.values(list).map((e) => e.entry),
+      { configDir }
+    )
+  ).resolves.toMatchInlineSnapshot(`
+    Array [
+      Object {
+        "directory": ".",
+        "files": "*.js",
+        "importPathMatcher": /\\^\\\\\\.\\[\\\\\\\\/\\]\\(\\?:\\(\\?!\\\\\\.\\)\\(\\?=\\.\\)\\[\\^/\\]\\*\\?\\\\\\.js\\)\\$/,
+        "titlePrefix": "",
+      },
+    ]
+  `);
+});
+
+test('expands entries that only yield mixed files', async () => {
+  const configDir = '/configDir/';
+  const list = createList(
+    [
+      { entry: '*.@(mdx|ts)', result: ['/configDir/my-file.mdx', '/configDir/my-file.ts'] },
+      { entry: '*.js', result: [] },
+    ],
+    configDir
+  );
+  glob.mockImplementation(createGlobMock(list));
+
+  await expect(
+    removeMDXEntries(
+      Object.values(list).map((e) => e.entry),
+      { configDir }
+    )
+  ).resolves.toMatchInlineSnapshot(`
+    Array [
+      Object {
+        "directory": ".",
+        "files": "**/my-file.ts",
+        "titlePrefix": "",
+      },
+      Object {
+        "directory": ".",
+        "files": "*.js",
+        "importPathMatcher": /\\^\\\\\\.\\[\\\\\\\\/\\]\\(\\?:\\(\\?!\\\\\\.\\)\\(\\?=\\.\\)\\[\\^/\\]\\*\\?\\\\\\.js\\)\\$/,
+        "titlePrefix": "",
+      },
+    ]
+  `);
+});
+
+test('passes titlePrefix', async () => {
+  const configDir = '/configDir/';
+  const list = createList(
+    [
+      {
+        entry: { files: '*.@(mdx|ts)', directory: '.', titlePrefix: 'foo' },
+        result: ['/configDir/my-file.mdx', '/configDir/my-file.ts'],
+      },
+    ],
+    configDir
+  );
+  glob.mockImplementation(createGlobMock(list));
+
+  await expect(
+    removeMDXEntries(
+      Object.values(list).map((e) => e.entry),
+      { configDir }
+    )
+  ).resolves.toMatchInlineSnapshot(`
+    Array [
+      Object {
+        "directory": ".",
+        "files": "**/my-file.ts",
+        "titlePrefix": "foo",
+      },
+    ]
+  `);
+});
+
+test('expands to multiple entries', async () => {
+  const configDir = '/configDir/';
+  const list = createList(
+    [
+      {
+        entry: { files: '*.@(mdx|ts)', directory: '.', titlePrefix: 'foo' },
+        result: [
+          '/configDir/my-file.mdx',
+          '/configDir/my-file1.ts',
+          '/configDir/my-file2.ts',
+          '/configDir/my-file3.ts',
+        ],
+      },
+    ],
+    configDir
+  );
+  glob.mockImplementation(createGlobMock(list));
+
+  await expect(
+    removeMDXEntries(
+      Object.values(list).map((e) => e.entry),
+      { configDir }
+    )
+  ).resolves.toMatchInlineSnapshot(`
+    Array [
+      Object {
+        "directory": ".",
+        "files": "**/my-file1.ts",
+        "titlePrefix": "foo",
+      },
+      Object {
+        "directory": ".",
+        "files": "**/my-file2.ts",
+        "titlePrefix": "foo",
+      },
+      Object {
+        "directory": ".",
+        "files": "**/my-file3.ts",
+        "titlePrefix": "foo",
+      },
+    ]
+  `);
```

Comment:
@kasperpeulen If you wanted to improve the mechanism, by rewriting the `files` property to not include `|mdx` or `mdx|`; this test would break. But you can use this test to TDD that feature.